woocommerce/woocommerce-ios
# Dynamic Dashboard: Coupons Card UI

Part of: #12716


## Description


Adds a first iteration of UI related code for the "Most Active Coupons" dashboard card. 

Changes
- View and View model for  "Most Active Coupons" dashboard card.
- Way to store and retrieve selected time range to App settings.
- Open the coupons list screen.
- Update coupon summary text format to match new design.

## Testing instructions


- Enable the feature flag `dynamicDashboardM2` and build the app.
- Confirm that the dashboard still has the same layout as before.
- Tap Edit button and confirm that there's an option for the most active coupons card and it's disabled.
- Enable coupons card and save the change.
- Confirm that the coupons card is displayed on the dashboard.
- Confirm that the UI matches the Figma design. 
- Tap View all coupons, confirm that coupons list screen is displayed.
- Tap the calendar icon and change time range, validate that time range selection works.
    - "All time" time range is removed as we don't have option to fetch that time range from API.
- Tap the ellipsis menu on the coupons card, confirm that there's an option Hide Coupons. 
- Tap Hide coupons, confirm that the customize screen is displayed.

Disable the `dynamicDashboardM2` flag and confirm that the Coupons card is not available.

## Screenshots


| Light | Dark |
|--------|--------|
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-14 at 22 39 09](https://github.com/woocommerce/woocommerce-ios/assets/524475/fd52030e-94b0-4e41-946e-2b7fd18c0094) | ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-14 at 22 38 57](https://github.com/woocommerce/woocommerce-ios/assets/524475/be39444e-a010-43d6-8a51-c3bf940c16d6) | 

---
